lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,491 @@
+          - name: ansible_user
+          - name: ansible_vmware_user
+        required: True
+      vmware_password:
+        description:
+          - Password for the connection.
+        env:
+          - name: VI_PASSWORD
+          - name: VMWARE_PASSWORD
+        vars:
+          - name: ansible_password
+          - name: ansible_vmware_password
+        required: True
+      vmware_port:
+        description:
+          - Port for the connection.
+        env:
+          - name: VI_PORTNUMBER
+          - name: VMWARE_PORT
+        vars:
+          - name: ansible_port
+          - name: ansible_vmware_port
+        required: False
+        default: 443
+      validate_certs:
+        description:
+          - Verify SSL for the connection.
+          - "Note: This will validate certs for both C(vmware_host) and the ESXi host running the VM."
+        env:
+          - name: VMWARE_VALIDATE_CERTS
+        vars:
+          - name: ansible_vmware_validate_certs
+        default: True
+        type: bool
+      silence_tls_warnings:
+        description:
+          - Don't output warnings about insecure connections.
+        vars:
+          - name: ansible_vmware_silence_tls_warnings
+        default: True
+        type: bool
+      vm_path:
+        description:
+          - VM path absolute to the connection.
+          - "vCenter Example: C(Datacenter/vm/Discovered virtual machine/testVM)."
+          - "ESXi Host Example: C(ha-datacenter/vm/testVM)."
+          - Must include VM name, appended to 'folder' as would be passed to M(vmware_guest).
+          - Needs to include C(vm) between the Datacenter and the rest of the VM path.
+          - Datacenter default value for ESXi server is C(ha-datacenter).
+          - C(vm) is not visible in vSphere Web Client but necessary for VMware API to work.
+        vars:
+          - name: ansible_vmware_guest_path
+        required: True
+      vm_user:
+        description:
+          - VM username.
+        vars:
+          - name: ansible_user
+          - name: ansible_vmware_tools_user
+        required: True
+      vm_password:
+        description:
+          - Password for the user in guest operating system.
+        vars:
+          - name: ansible_password
+          - name: ansible_vmware_tools_password
+        required: True
+      exec_command_sleep_interval:
+        description:
+          - Time in seconds to sleep between execution of command.
+        vars:
+          - name: ansible_vmware_tools_exec_command_sleep_interval
+        default: 0.5
+        type: float
+      file_chunk_size:
+        description:
+          - File chunk size.
+          - "(Applicable when writing a file to disk, example: using the C(fetch) module.)"
+        vars:
+          - name: ansible_vmware_tools_file_chunk_size
+        default: 128
+        type: integer
+      executable:
+        description:
+            - shell to use for execution inside container
+        default: /bin/sh
```

Comment:
Just a suggestion. We can rely on the GuestID here to set the executable: If the GuestId starts with "win" -> PowerShell, otherwise, /bin/sh.

Reply:
I think this is handled by setting ansible_shell_type=powershell

Reply:
The plugin already hardcodes it to "cmd.exe" if the guestfamily is windows.